packages/clay-charts/src/ChartBase.js

```diff
@@ -731,21 +763,52 @@ ChartBase.STATE = {
 	 * @memberof ChartBase
 	 * @type {?Array|undefined}
 	 * @default undefined
+	 * @deprecated since 2.0.0-rc.2
 	 */
-	columns: Config.arrayOf(
-		Config.shapeOf({
-			axis: Config.oneOf(['y', 'y2']),
-			class: Config.string(),
-			color: Config.string(),
-			data: Config.array().required(),
-			hide: Config.bool(),
-			id: Config.required().string(),
-			name: Config.string(),
-			regions: Config.array(),
-			type: Config.oneOf(types.all),
-			x: Config.string(),
-		})
-	),
+	columns: Config.oneOfType([
+		Config.arrayOf(
+			Config.shapeOf({
+				axis: Config.oneOf(['y', 'y2']),
+				class: Config.string(),
+				color: Config.string(),
+				data: Config.array().required(),
+				hide: Config.bool(),
+				id: Config.required().string(),
+				name: Config.string(),
+				regions: Config.array(),
+				type: Config.oneOf(types.all),
+				x: Config.string(),
+			})
+		),
+		Config.func(),
```

Comment:
I don't think we want to allow the other types for the old API. `columns` should accept only the old column shape

Reply:
Yes it make sense